tv_apps/tv_shared/components/smart-dialog/style.css

```diff
@@ -0,0 +1,99 @@
+/**
+ * Smart-Dialog
+ *   - State:
+ *       1. closed
+ *       2. opening
+ *       3. opened - content visible
+ *       4. closing
+ * Default scale: 1920x1080 (2*sqrt(960*960 + 1080*1080) / 20 = 144.5)
+ */
+
+smart-dialog {
+  position: fixed;
+  z-index: 2147483647;
+  height: 2rem;
+  width: 2rem;
+  top: calc(100% - 1rem);
+  left: calc(50% - 1rem);
+  background-color: rgba(0, 0, 0, 0.9);
+  transition: none;
+  transform: scale(20);
+}
+
+smart-dialog:before {
+  content: attr(title);
+  display: block;
+  margin-top: 5rem;
+  margin-left: 7.6rem;
+  font-family: "Fira Sans";
+  font-size: 5.4rem;
+  font-style: italic;
+  color: #ffffff;
+}
+
+smart-dialog > * {
+  display: none;
+}
+
+smart-dialog.closed {
+  visibility: hidden;
+}
+
+smart-dialog.opened {
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  transform: scale(1);
+}
+
+smart-dialog.opened > * {
+  display: block;
+}
+
+smart-dialog.opening {
+  border-radius: 50%;
+  transform: scale(144.5);
+  /* transitionDuration should be the same as openDuration variable in script */
+  transition: transform 1s cubic-bezier(0, 0, 0 ,1);
+}
+
+smart-dialog.prepare-close {
+  transform: scale(144.5);
+}
+
+smart-dialog.closing {
+  border-radius: 50%;
+  /* transitionDuration should be the same as closeDuration variable in script */
+  transition: transform 0.65s cubic-bezier(0.6, 0, 1 ,1);
+}
+
+/* 720x480 */
+@media (max-width: 720px) {
+  smart-dialog {
+    transform: scale(13.3);
+  }
+
+  smart-dialog.opening {
+    transform: scale(60);
+  }
+
+  smart-dialog.readyToClose {
+    transform: scale(60);
+  }
+}
+
+/* 1280x800 */
+@media (min-width: 720px) and (max-width: 1280px) {
```

Comment:
For flatfish, we may have portrait/landscape mode. Once user rotate the device, the width may be 800 and height may be 1280. Although it is covered by this rule, we still need to include the height condition here. The main reason is that we may have 1080 as width once we rotated a full HD TV which supports rotation.